### DISCUSSION TITLE
Judging antenna quality / metrics for automated optimization

### DISCUSSION THREAD

**Discussion by oberstet** [thliebig/openEMS-Project#143] [Q&A Usage]
Hi all, I'm a RF newbie and I'd like to design an antenna, so a steep learning curve is expected, but after some days fiddling around I now finally got all the basics (geometry, meshes, etc) working and plots working, and now I want to go to actual optimization.

For that, I need a metric as an objective function for automated optimization, however (lacking experience / know-how), I'm not sure _what_ exactly to optimize for.

Here is a plot from an antenna model I "hand-tuned" which illustrates my best guess of what might work. Any help, pointing in direction, feedback and help would be highly appreciated.

![grafik](https://github.com/thliebig/openEMS-Project/assets/233340/8eacff42-602d-4343-846f-4c2929e62647)

----

I'd target the box marked green, but this only covers S11 reflection in dB.

Do I need to also target something else besides S11?

Like impedance? If so, should the target be a uniform value for the whole frequency range f1 - f2 I want to cover?

Here is a value I believe should be good?

```
Found resonance frequency at 869 MHz with -47 dB at 121 Ohm
```

IOW: if I can find an optimization algo that can tune for a metric that "aggregates" all relevant aspects, the rest is just some programming work which I can figure out.

Is -10dB good enough, I want to use this in the whole 500MHz range, is it "flat enough"? The border frequencies f1/f2 are of "special interest", hence it is actually desirable to have low S11 there specifically.

Also, in my picture above, beyond 950MHz things seem to go wild (I don't care much), S12 going significantly positive, what does that mean actually? Should I be worried? Again, the freq range I'm interested in is lower.

So my question really is about the RF parts.

Again, thanks so much for any help!, Cheers, /Tobias

**Comment by LubomirJagos** [comment]
that part over 800MHz is really strange, like S11 should never be higher than 0dB since it means there is somewhere energy generated in model, it could be due wrong gridlines position and due some numerical instabilities
I would suggest to put into model E monitor to see field propagation over time and also set frequency range from 300-900MHz

**Reply by oberstet** [reply]
awesome, thanks! I will try what you suggest ..

**Reply by LubomirJagos** [reply]
and as end criteria put there at least 0.001 what stops simulation when energy lower under -30dB, and don't forget to set boundary conditions to some absorbing material PML_8 is best, but MUR could be also used

**Reply by oberstet** [reply]
I was using sth like this .. not sure anymore which one exactly (the energy cutoff) ... I need to hook up a database, otherwise I'm going insane .. too many charts and numbers;) anyways, here:

```python
# FDTD = openEMS(EndCriteria=1e-4)
FDTD = openEMS(EndCriteria=6e-4)
# FDTD = openEMS(EndCriteria=1e-3)

FDTD.SetGaussExcite(f0, fc)
FDTD.SetBoundaryCond(['MUR', 'MUR', 'MUR', 'MUR', 'MUR', 'PML_8'])
```

I don't actually know what MUR and PML_8 means ... I just saw that in the tutorials. Is this MUR/PML_8 choice good/sane?

**Reply by oberstet** [reply]
here is another of the tons of charts I generated (definitely has the 3 charts corresponding to 1 try/setup):

![grafik](https://github.com/thliebig/openEMS-Project/assets/233340/6cd01dbf-d7ef-4b4c-9cc3-b7b889edaae7)

**Reply by LubomirJagos** [reply]
hmmm, I'm not expert but what I read from documentation:
    - MUR is 1 cell thick boundary condition absorbing all energy but it's some kind of mathematic construct which should be faster, I had similar problem like you with instability when I used it once
    - PML_8, PML_4 or PML_n will take n cells thick layer for boundary condition and it's proper model of absorbing energy, due to mathematical model it is slowing simulation but it's proper model
    
in your case when you don't know choose PML even at cost slow simulation but you will have no problem with some instabilities and will see hopefully it will solve that positive S11 as positive S11 means there is energy floating into your model what should be visible from E field monitor you will see like your model start lighting red after some time, I would suggest to put all boundaries PML and put E field to see what is happening

I know that this is not best kind of advice what you can get, it's just way what I will do, maybe somebody here give you better advice, but I suppose you are now running your simulations so at the moment try this, at this point it's really usefull to visualize fields in your model and in paraview it really gives you good insight what is happening there, normally you should see energy floating along your antenna and energy should be radiated and absorbed at boundaries, if it doesn't feel good after you visalize it there are just 2 options:
    1. your model is really bad in engineering design terms and error what you see in graphs is real error in your model (but since there is some pass band it seems like your model somehow runs)
    2. your model has simulation errors - gridlines on metal edges, gridlines infinately too close each others :D numerical errors
    
but I think that you are on good way, but not at the end ;)

**Reply by oberstet** [reply]
> 2. your model has simulation errors - gridlines on metal edges, gridlines infinately too close each others :D numerical errors

yeah. that would also be my best guess. like mesh size not fine enough for the high end of the spectrum used leading to "interesting" (unreal) effects when simulating.

for my eyes, it's popping out: why the wild behavior to the right of spectrum? must be sth "meta" (aka model setup) related, not really real ... maybe that's also my over-interpretation .. having a math background makes you spot and hate any asymmetries;)

**Comment by oberstet** [comment]
> I would suggest to put all boundaries PML and put E field to see what is happening

_all_ of them? and all of them PML_8?

I was (in the figures posted) already using MUR in 5 of 6 items in the `FDTD.SetBoundaryCond` call ( it wants 6 .. which I would guess it's the directions along the 3 cartesian axis?), and my best guest for the last one in 

```python
FDTD.SetBoundaryCond(['MUR', 'MUR', 'MUR', 'MUR', 'MUR', 'PML_8'])
```

would be: that's z-axis to the negative - the space _below_ the groundplane. but this is a wild guess of mine. I simply don't know, have no real clue;)

anyways. I need to figure out and add this E-field stuff .. haven't used it before .. np, should be able to get sth going. I feel like I am close to having my toolbox and all elements for simulation/model complete .. and then I can finally move forward to what I actually want ... optimize the dual antenna ... then PCB ... and _then_ I can go back to where I feel more at home: code and firmware;)

**Reply by oberstet** [reply]
*how should I position and size the E-field monitor?*

----

fwiw, I asked chatgpt - below is the answer, does that make sense?

![grafik](https://github.com/thliebig/openEMS-Project/assets/233340/e7b07449-d929-4581-8329-a3dffb97d3ef)

and

![grafik](https://github.com/thliebig/openEMS-Project/assets/233340/bced089d-431c-4861-9cf1-22be92bccbd3)
